tools/make.py

```diff
@@ -207,14 +207,20 @@
 
     # Target
     if options.mcu is None :
-        args_error(parser, "[ERROR] You should specify an MCU")
+        args_error(parser, "argument -m/--mcu is required")
     mcu = options.mcu[0]
 
     # Toolchain
     if options.tool is None:
-        args_error(parser, "[ERROR] You should specify a TOOLCHAIN")
+        args_error(parser, "argument -t/--tool is required")
     toolchain = options.tool[0]
 
+    if (options.program is None) and (not options.source_dir):
+        args_error(parser, "one of -p, -n, or --source is required")
+
+    if options.source_dir and not options.build_dir:
+        args_error(parser, "argument --build is required when argument --source is provided")
```

Comment:
Yes, We do. I agree, it does feel a bit weird. If you don't specify a build directory, you get a traceback about None not having startswith, I think.